### ISSUE TITLE
sanctified way to know if IP stands for an IPv4 or IPv6 ?

### ISSUE THREAD

**Issue by teto** [andrewthad/haskell-ip#54]
I need to send the address family (AF_INET or AF_INET6) through netlink. 
I have all my IPs recorded with the type IP. How can I retrieve from the data IP the type of the IP ? IPv4 or IPv6 ?

**Comment by chessai** [comment]
Use Net.IP.case_:

```haskell
isIPv4 :: IP -> Bool
isIPv4 = case_ (const True) (const False)

isIPv6 :: IP -> Bool
isIPv6 = case_ (const False) (const True)
```

It's possible that these could be added as helpers to this module, since
this seems like a common enough thing to want to do.

**Comment by chessai** [comment]
Use Net.IP.case_:

```haskell
isIPv4 :: IP -> Bool
isIPv4 = case_ (const True) (const False)

isIPv6 :: IP -> Bool
isIPv6 = case_ (const False) (const True)
```

It's possible that these could be added as helpers to this module, since
this seems like a common enough thing to want to do.

**Comment by teto** [comment]
thanks !
Yeah having these upstreamed makes sense IMO.

**Comment by chessai** [comment]
I just released 1.5.1 which adds these.

**Comment by andrewthad** [comment]
Thanks for adding these.  While I’m thinking about it, this situation is somewhere where pattern synonyms would shine.  They would complement the existing  functions rather than replace them.
